tests/jsunit/event_test.js

```diff
@@ -573,3 +671,75 @@ function helper_addMoveEventParent(events, block, parent) {
   block.setParent(parent);
   events[events.length-1].recordNew();
 }
+
+function test_events_newblock_newvar() {
+  eventTest_setUpWithMockBlocks();
+
+  Blockly.Events.fire = temporary_fireEvent;
+  temporary_fireEvent.firedEvents_ = [];
+  // Expect three calls to genUid: one to set the block's ID, one for the event
+  // group's id, and one for the variable's ID.
+  setUpMockMethod(mockControl_, Blockly.utils, 'genUid', null, ['1', '2', '3']);
+  try {
+    var block = workspace.newBlock('field_variable_test_block');
+
+    var firedEvents = workspace.undoStack_;
+    // Expect two events: varCreate and block create.
+    assertEquals(2, firedEvents.length);
```

Comment:
Prefer to use assert messages (optional leading argument: http://www.jsunit.net/documentation/assertions/) instead of code comments.  Using a assert message will expose the expectation in the test logs upon failure.